deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.java

```diff
@@ -319,7 +319,20 @@ private void generateDeepLinkDelegate() throws IOException {
         .beginControlFlow("if (activity == null)")
         .addStatement("throw new $T($S)", NullPointerException.class, "activity == null")
         .endControlFlow()
-        .addStatement("$T sourceIntent = activity.getIntent()", ANDROID_INTENT)
+        .addStatement("return dispatchFrom(activity, activity.getIntent())")
+        .build();
+
+    MethodSpec dispatchFromMethodWithIntent = MethodSpec.methodBuilder("dispatchFrom")
+        .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+        .returns(DEEPLINKRESULT)
+        .addParameter(ClassName.get("android.app", "Activity"), "activity")
+        .addParameter(ClassName.get("android.content", "Intent"), "sourceIntent")
+        .beginControlFlow("if (activity == null)")
+        .addStatement("throw new $T($S)", NullPointerException.class, "activity == null")
```

Comment:
I just realized that if I remove 319 then 322 could null out. So while it is a redundant check it is needed for good error reporting. Can still remove it if you want but I don't think we do.